api/Controllers/RoomController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using System.Linq;
+using System.Threading.Tasks;
+using iTechArt.Hotels.Api.Entities;
+using iTechArt.Hotels.Api.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.EntityFrameworkCore;
+
+namespace iTechArt.Hotels.Api.Controllers
+{
+    [Route("api/hotels")]
+    [ApiController]
+    public class RoomController : Controller
+    {
+        private readonly HotelsDatabaseContext _hotelsDb;
+        private readonly IMapper _mapper;
+
+        public RoomController(
+            HotelsDatabaseContext hotelsDb,
+            IMapper mapper
+        )
+        {
+            _hotelsDb = hotelsDb;
+            _mapper = mapper;
+        }
+
+        [Route("{hotelId}/rooms/{roomId}")]
+        [HttpDelete]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> DeleteRoom([FromRoute] int hotelId, [FromRoute] int roomId)
+        {
+            if (await GetHotelEntityAsync(hotelId) == null)
+            {
+                return BadRequest("Such hotel does not exist");
+            }
```

Comment:
As you not using the Hotel entity that GetHotelEntityAsync returns can we make some improvements here?